File: src/client/flutter/lib/models/entities/choice.dart
import 'package:flutter/material.dart';

import 'my_color.dart';

class Choice {
  int id;
  String name;
  int idVote;
  String? description;
  String? candidateNIR;
  String candidate;
  int? idColor;
  MyColor? color;

  Choice({
    required this.id,
    required this.name,
    required this.idVote,
    this.description,
    this.candidateNIR,
    this.candidate = "",
    this.idColor
  });

  Choice.fromJson(Map<String, dynamic> json)
      : id = json["id"],
        name = json["name"],
        idVote = json["id_vote"],
        description = json["description"],
        candidateNIR = json["candidat_nir"],
        idColor = json["idColor"],
        candidate = json["candidat"] {
    if(json["color"] != null) {
      color = MyColor.fromJson(json["color"] as Map<String, dynamic>);
    }
  }

  Object toJson() {
    Map<String, dynamic> object = {
      "name": name,
      "description": description,
    };
    if (candidateNIR != null) {
      object["candidat_nir"] = candidateNIR;
    }
    if (idColor != null) {
      object["id_color"] = idColor.toString();
    }
    return object;
  }

  Color getColor() {
    if(color == null) {
      return Colors.grey;
    } else {
      return color!.toColor();
    }
  }

}

File: src/app/android/app/src/main/java/com/cyberacy/app/models/entities/Choice.kt
package com.cyberacy.app.models.entities

import android.graphics.Color
import com.cyberacy.app.R
import com.cyberacy.app.models.services.ApiConnection
import com.google.gson.annotations.SerializedName
import retrofit2.HttpException
import retrofit2.await

class Choice(

    @SerializedName("id")
    val id: Int,

    @SerializedName("name")
    val name: String,

    @SerializedName("id_vote")
    val idVote: Int,

    @SerializedName("description")
    val description: String,

    @SerializedName("candidat_nir")
    val candidateNIR: String?,

    @SerializedName("candidat")
    val candidate: String?,

    @SerializedName("id_color")
    val idColor: Int?,

    @SerializedName("color")
    val color: MyColor?,

) {

    fun getColorChoice(): Int {
        return color?.toColor() ?: Color.argb(255, 200, 200, 200)
    }

    companion object Service {

        suspend fun getChoices(idVote: Int, idRound: Int): List<Choice> {
            try {
                val response: ResponseAPI<List<Choice>?> =
                    ApiConnection.connection().getChoicesFromRound(idVote, idRound).await()
                return response.data ?: emptyList()
            } catch (e: HttpException) {
                throw e
            }
        }

    }

}